speculative_decoding/model/model.py

```diff
@@ -47,43 +48,45 @@ def __init__(self, **kwargs):
             raise NotImplementedError("openai-compatible")
 
     def load(self):
-        tensor_parallel_count = self._config["model_metadata"].get(
-            "tensor_parallelism", 1
-        )
-        pipeline_parallel_count = self._config["model_metadata"].get(
-            "pipeline_parallelism", 1
-        )
+        if not helpers.is_triton_server_alive():
+            tensor_parallelism = self._config["model_metadata"].get(
+                "tensor_parallelism", 1
+            )
+            pipeline_parallelism = self._config["model_metadata"].get(
+                "pipeline_parallelism", 1
+            )
 
-        env = {}
-        try:  # Secrets does not have __contains__.
-            hf_access_token = self._secrets["hf_access_token"]
-            env["HUGGING_FACE_HUB_TOKEN"] = hf_access_token
-        except Exception:  # `SecretNotFound` from truss template cannot be referenced.
-            hf_access_token = None
-
-        # Target model.
-        huggingface_hub.snapshot_download(
-            self._config["model_metadata"]["engine_repository"],
-            local_dir=os.path.join(TRITON_DIR, TARGET_MODEL_KEY, "1"),
-            local_dir_use_symlinks=False,
-            max_workers=4,
-            use_auth_token=hf_access_token,
-        )
-        # Draft model.
-        huggingface_hub.snapshot_download(
-            self._config["model_metadata"]["speculative_decoding"][
-                "draft_engine_repository"
-            ],
-            local_dir=os.path.join(TRITON_DIR, DRAFT_MODEL_KEY, "1"),
-            local_dir_use_symlinks=False,
-            max_workers=4,
-            use_auth_token=hf_access_token,
-        )
+            env = {}
+            try:  # Secrets does not have __contains__.
+                hf_access_token = self._secrets["hf_access_token"]
+                env["HUGGING_FACE_HUB_TOKEN"] = hf_access_token
+            except (
+                Exception
+            ):  # `SecretNotFound` from truss template cannot be referenced.
+                hf_access_token = None
```

Comment:
May be extract out into a function:
`def supplied_hf_token(self) -> Optional[str]:`